### PR TITLE
Add ability to patch service accounts upfront through create env

### PR DESCRIPTION
**Pull request by a-roberts**
See the issue https://github.com/jenkins-x/jx/issues/1857 for background

I would like to add a few tests as well: 
- can it handle no secrets correctly
- can it handle one
- can it handle many

Creating the PR here so folks can see the code and what we are thinking: 
credits to @johnmcollier for the patching code, John is interested in doing the same for dev pods.

```
build/jx create env --pull-secrets="foo bar bas"
... extra info omitted for brevity ...

Created environment kfc
Namespace jx-kfc created
 Service account "default" in namespace "jx-kfc" configured to use pull secret(s) [foo bar bas]
Pull secret(s) must exist in namespace jx-kfc before deploying your applications in this environment
Created Jenkins Project: http://jenkins.jx.9.20.201.31.nip.io/job/a-roberts/job/environment-jx-kfc/

Note that your first pipeline may take a few minutes to start while the necessary images get downloaded!

Creating GitHub webhook for a-roberts/environment-jx-kfc for url http://jenkins.jx.9.20.201.31.nip.io/github-webhook/
```

I verified the service account in the target namespace is configured to use all the pull secrets. This is my first implementation PR so let's see if the tests work here!

### CODE REVIEW

File: pkg/jx/cmd/common.go [hunk 1]
```diff
@@ -285,6 +289,11 @@ func (o *CommonOptions) TeamAndEnvironmentNames() (string, string, error) {
 	return kube.GetDevNamespace(kubeClient, currentNs)
 }
 
+func (o *CommonOptions) ParseImagePullSecrets() []string {
+	pullSecrets := strings.Fields(o.PullSecrets)
+	return pullSecrets
+}
+
```
Comment (pmuir) [inline pkg/jx/cmd/common.go]: Should be GetImagePullSecrets() to be consistent with other method names

Comment (a-roberts) [reply pkg/jx/cmd/common.go]: Agreed, thanks

Comment (pmuir) [reply pkg/jx/cmd/common.go]: Nice. Unortunately that's as far as I can provide a sensible review ;-)

**Comment by a-roberts** [comment]
Hi there, anything I need to do for this? Updating branch every so often but don't want to forever

**Comment by rawlingsj** [comment]
My slight concern with this is we're updating the default service account in a namespace.  I guess I'm wondering what happens if a pod uses a specific serviceaccount and not default?  Also should we instead be creating a new serviceaccount and updating that?  Just trying to think through this PR.

**Comment by jstrachan** [comment]
happy to merge and keep improving - but yeah its probably better to have a SA per environment and use that instead of the `default`?

**Comment by rawlingsj** [comment]
OK lets merge and update a separate SA in another PR.

**Comment by a-roberts** [comment]
Thanks all, yeah very valid concerns, when creating an environment I noticed several service accounts actually, and if the pods are being run under the service account "default" it made sense to patch this. 

Agreed, it may be better to create a different service account that we patch, and then we pass in this service account name as part of the pod template for pipelines (so that we know the pipeline gets run under that service account, which has access to the registry for pushing to). 

Deployed applications in this environment would need to run under the same service account to have that secret as well to pull the image. So it would be a dual-use service account really, perhaps "pipeline-sa" would be a useful name for it.

**Pull request merged**